packages/framework-types/src/config.ts

```diff
@@ -102,6 +105,11 @@ export class BoosterConfig {
   }
 
   public get provider(): ProviderLibrary {
+    if (!this._provider && this.providerPackage) {
+      const rockets = this.rockets ?? []
+      const provider = require(this.providerPackage)
+      this._provider = provider.Provider(rockets)
+    }
```

Comment:
Here's where the magic happens. The rest of the codebase stays the same, because when accessing `config.provider` as a getter, this code will get trigger, and if the provider is not set, it will `require` by using the `config.providerPackage` and `config.rockets` fields

Reply:
I like it!